File: documentation/overview/webhooks/sample.js
// Sample code in Node.js

const crypto = require('crypto');
// webhook_payload => {serialized_response: <encrypted_json_string>}
const str = <encrypted_json_string>;
const key = <webhook_encryption_key>;

const algorithm = 'aes-128-cbc';
try {
    const iv = str.substring(0,16);
    const encry_data = str.substring(16, ) ;
    const decipher = crypto.createDecipheriv(algorithm, key, iv);
    decipher.setAutoPadding(true);
    let decry_data = decipher.update(encry_data, 'base64', 'utf8');
    decry_data += decipher.final('utf8');
    console.log(decry_data);

    // parse json string to obj
    let event_obj = JSON.parse(decry_data);
    console.log(event_obj);
}
catch(err) { }

File: documentation/overview/webhooks/sample.rb
// Sample code in Ruby

require 'base64'
require 'securerandom'
require 'openssl'

# webhook_payload => {serialized_response: <encrypted_json_string>}
str = <encrypted_json_string>
key = <webhook_encryption_key>

iv = str[0..15]
encry_data = Base64.decode64(str[16..-1])
decipher = OpenSSL::Cipher::AES.new(128, :CBC)
decipher.decrypt
decipher.key = key
decipher.iv = iv
decry_data = decipher.update(encry_data) + decipher.final

# parse json string to obj
event_obj = JSON.parse(decry_data)
